plugins/basic/nexus-repository-nuget/src/main/java/com/sonatype/nexus/repository/nuget/internal/NugetGalleryFacetImpl.java

```diff
@@ -377,24 +393,40 @@ public InputStream get() throws IOException {
   }
 
   @Override
-  public DateTime getLastUpdatedDate(final String id, final String version) {
+  public DateTime getLastVerified(final String id, final String version) {
     checkNotNull(id);
     checkNotNull(version);
 
     try (StorageTx tx = openStorageTx()) {
-      Component component = findComponent(tx, id, version);
-      if (component == null) {
-        return null;
-      }
-      Asset asset = tx.firstAsset(component);
+      final Asset asset = findAsset(tx, id, version);
 
-      final NestedAttributesMap nugetAttributes = asset.formatAttributes();
-      Date date = nugetAttributes.get(P_LAST_UPDATED, Date.class);
+      Date date = asset.formatAttributes().get(P_LAST_VERIFIED_DATE, Date.class);
+
+      log.info("Content for {} {} last verified as of {}", id, version, date);
       return new DateTime(checkNotNull(date));
     }
   }
 
   @Override
+  public void setLastVerified(final String id, final String version, final DateTime date) {
+    checkNotNull(date);
+    final Date priorDate = storage.perform(new Operation<Date>()
+    {
+      @Override
+      public Date execute(final StorageTx tx) {
+        Asset asset = findAsset(tx, id, version);
+        checkState(asset != null);
+
+        final Date priorDate = asset.formatAttributes().get(P_LAST_VERIFIED_DATE, Date.class);
+        asset.formatAttributes().set(P_LAST_VERIFIED_DATE, date.toDate());
+
+        return priorDate;
+      }
+    });
+    log.info("Updating last verified date of {} {} from {} to {}", id, version, priorDate, date);
```

Comment:
Is this going to happen a lot - if so should it be debug rather than info? Ditto for the logging in `getLastVerified`